x/superfluid/keeper/stake_test.go

```diff
@@ -1188,6 +1192,173 @@ func (suite *KeeperTestSuite) TestPartialSuperfluidUndelegateToConcentratedPosit
 	}
 }
 
+func (s *KeeperTestSuite) TestLockExistingFullRangePositionAndSFStake() {
+	testCases := []struct {
+		name                        string
+		positionId                  uint64
+		lowerTick                   int64
+		upperTick                   int64
+		superfluidNotEnabledOnDenom bool
+		expectedError               error
+	}{
+		{
+			name:       "happy path",
+			positionId: 2,
+			lowerTick:  cltypes.MinInitializedTick,
+			upperTick:  cltypes.MaxTick,
+		},
+		{
+			name:          "position ID doesn't exist",
+			positionId:    3,
+			lowerTick:     cltypes.MinInitializedTick,
+			upperTick:     cltypes.MaxTick,
+			expectedError: cltypes.PositionIdNotFoundError{PositionId: 3},
+		},
+		{
+			name:                        "underlying cl denom is not enabled for sfs",
+			positionId:                  2,
+			lowerTick:                   cltypes.MinInitializedTick,
+			upperTick:                   cltypes.MaxTick,
+			superfluidNotEnabledOnDenom: true,
+			expectedError:               errorsmod.Wrapf(types.ErrNonSuperfluidAsset, "denom: %s", "cl/pool/2"),
+		},
+		{
+			name:          "position lower tick is not full range",
+			positionId:    2,
+			lowerTick:     cltypes.MinInitializedTick + 1,
+			upperTick:     cltypes.MaxTick,
+			expectedError: cltypes.PositionNotFullRangeError{PositionId: 2, LowerTick: cltypes.MinInitializedTick + 1, UpperTick: cltypes.MaxTick},
+		},
+		{
+			name:          "position upper tick is not full range",
+			positionId:    2,
+			lowerTick:     cltypes.MinInitializedTick,
+			upperTick:     cltypes.MaxTick - 1,
+			expectedError: cltypes.PositionNotFullRangeError{PositionId: 2, LowerTick: cltypes.MinInitializedTick, UpperTick: cltypes.MaxTick - 1},
+		},
+	}
+
```

Comment:
can we have test case for sender not the owner of position ?

Reply:
non owner test case added here https://github.com/osmosis-labs/osmosis/pull/5745/commits/d3c1b0fb39150a80643df73f7d4f0977ebe0628a